tests/www/views/test_views.py

```diff
@@ -296,6 +297,173 @@ def get_task_instance(session, task):
         assert dagrun.get_state() == State.QUEUED
 
 
+@pytest.mark.parametrize(
+    "downstream, set_state, end_state",
+    [
+        (
+            True,
+            [
+                ("task_1", State.SUCCESS),
+                ("task_2", State.SUCCESS),
+                ("task_3", State.SUCCESS),
+            ],
+            [
+                ("task_1", State.FAILED),
+                ("task_2", State.FAILED),
+                ("task_3", State.FAILED),
+            ],
+        ),
+        (
+            True,
+            [
+                ("task_1", State.SUCCESS),
+                ("task_2", State.SUCCESS),
+                ("task_3", State.FAILED),
+            ],
+            [
+                ("task_1", State.FAILED),
+                ("task_2", State.FAILED),
+                ("task_3", State.FAILED),
+            ],
+        ),
+        (
+            False,
+            [
+                ("task_1", State.SUCCESS),
+                ("task_2", State.SUCCESS),
+                ("task_3", State.FAILED),
+            ],
+            [
+                ("task_1", State.FAILED),
+                ("task_2", State.SUCCESS),
+                ("task_3", State.NONE),
```

Comment:
Why does this one become none?

Reply:
This is to ensure the behavior in https://github.com/apache/airflow/pull/13037 is kept in place. I also find it slightly counter intuitive since downstream is not selected and marking a task as failure shouldn't clear the state of downstream tasks that are failed already for re-run. Should we change it so that failed state is preserved?